surveyor.py

```diff
@@ -5,477 +5,457 @@
-
-            # write any remaining results
-            for tag, nested_results in product.get_results().items():
-                _write_results(writer, nested_results, tag.tag, str(tag.data), tag, log)
-
-        # if there's sigma rules to be processed
-        if len(sigma_rules) > 0:
-            translated_rules = sigma_translation(product_str, sigma_rules)
-            if len(translated_rules['queries']) != len(sigma_rules):
-                log.warning(f"Only {len(translated_rules['queries'])} out of {len(sigma_rules)} were able to be translated.")
-            for rule in tqdm(translated_rules['queries'], desc="Processing sigma rules", disable=opt.no_progress):
-                program = f"{rule['title']} - {rule['id']}"
-                source = 'Sigma Rule'
-
-                product.nested_process_search(Tag(program, data=source), {'query': [rule['query']]}, base_query)
-
-                if product.has_results():
-                    # write results as they become available
-                    for tag, nested_results in product.get_results(final_call=False).items():
-                        _write_results(writer, nested_results, program, str(tag.data), tag, log,
-                                       use_tqdm=True)
-
-                    # ensure results are only written once
-                    product.clear_results()
-
-            # write any remaining results
-            for tag, nested_results in product.get_results().items():
-                _write_results(writer, nested_results, tag.tag, str(tag.data), tag, log)
-
-        if output_file:
-            log_echo(f"\033[95mResults saved: {output_file.name}\033[0m", log)
-    except KeyboardInterrupt:
-        log_echo("Caught CTRL-C. Exiting...", log)
-    except Exception as e:
-        log_echo(f'Caught {type(e).__name__} (see log for details): {e}', log, logging.ERROR)
-        log.exception(e)
-    finally:
-        if output_file:
-            output_file.close()
-
-
-def create_generic_product_command(name: str) -> Callable:
-    @click.pass_context
-    def command(ctx):
-        survey(ctx, name)
-
-    command.__name__ = name
-    return command
-
-
-# create click commands for all products that don't have a command function defined
-for product_name in get_products():
-    dir_res = dir()
-    if product_name not in dir_res:
-        cli.command(name=product_name, help=f'Query {product_name}')(create_generic_product_command(str(product_name)))
+                print(self._table_template_str.format(*row))
 
 if __name__ == "__main__":
-    cli()
+    
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--prefix", help="Output filename prefix.", type=str)
+    parser.add_argument("--profile", help="The credentials profile to use.", type=str)
+    parser.add_argument("--days", help="Number of days to search.", type=int)
+    parser.add_argument("--minutes", help="Number of minutes to search.", type=int)
```

Comment:
We can put `days` and `minutes` into a mutually exclusive group https://docs.python.org/3/library/argparse.html#mutual-exclusion